src/core/container.rs

```diff
@@ -62,15 +63,18 @@ where
         image: I,
         command: Command,
     ) -> Self {
+        let ports = docker_client.ports(&id);
+
         let container = Container {
             id,
             docker_client: Box::new(docker_client),
             image,
             command,
+            ports,
             client_lifetime: PhantomData,
         };
 
-        container.block_until_ready();
+        container.block_until_ready(container.image().ready_conditions());
```

Comment:
This is a weird indirection now. I think we may have to remove `container#block_until_ready` and instead put this functionality onto the `Docker` trait.